content/blog/quarkus-and-testcontainers.asciidoc

```diff
@@ -0,0 +1,142 @@
+---
+title: "Quarkus and Testcontainers"
+date: 2020-11-28T17:45:00+01:00
+draft: false
+---
+:source-highlighter: rouge
+:rouge-style: base16.dark
+:icons: font
+:imagesdir: /images
+ifdef::env-github[]
+:imagesdir: ../../static/images
+endif::[]
+
+The https://www.testcontainers.org/[Testcontainers] project is invaluable for spinning up containerized resources during your (JUnit) tests,
+e.g. databases or Kafka clusters.
+
+The project provides the https://www.testcontainers.org/quickstart/junit_5_quickstart/[`@Testcontainers`] JUnit extension for controlling the lifecycle of containers used by a test.
+When testing a https://quarkus.io/[Quarkus] application though, this is at odds somewhat with Quarkus' own https://quarkus.io/guides/getting-started-testing#recap-of-http-based-testing-in-jvm-mode[`@QuarkusTest`] extension;
+it's a best practice to avoid fixed ports for any containers started by Testcontainers.
```

Comment:
up to you, of course, but you may link my article that explains the "why" part:
https://bsideup.github.io/posts/testcontainers_fixed_ports/